x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/RegionMapChart/index.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo } from 'react';
+import { ChoroplethMap } from './ChoroplethMap';
+import { ColorProgressionBar } from './ColorProgressionBar';
+import { useAvgDurationByCountry } from '../../../../../hooks/useAvgDurationByCountry';
+
+function getProgressionColor(
+  scale: number,
+  // default to euiColorPrimary (#006BB4)
+  color = { hue: 204, saturation: 100, lightness: { min: 20, max: 90 } }
+) {
+  const { min, max } = color.lightness;
+  const lightness = Math.round(min + (max - min) * (1 - scale));
+  return `hsl(${color.hue},${color.saturation}%,${lightness}%)`;
+}
+
+const RegionMapChartToolTip: React.SFC<{
+  geojsonProperties: { [name: string]: any };
+  data?: { key: string; value: number } & Partial<{ docCount: number }>;
+}> = ({ geojsonProperties, data }) => {
+  return (
+    <div style={{ textAlign: 'center' }}>
```

Comment:
What's our policy re: inline styles?

Reply:
I haven't put too much thought into this but sometimes I also use inline styling instead of creating a separate "styled-components" component. For instance if I only need to add a single style, and the component is not re-used anywhere else. But perhaps it's better to go ahead and create the rule like we normally would? What do you think?

If we upgrade to the latest version of styled-components we'll get access to the `css` prop which feels like inline styling but is turned into a styled-component under the hood.
https://medium.com/styled-components/announcing-native-support-for-the-css-prop-in-styled-components-245ca5252feb